### PR TITLE
Use hooks for "create" tasks

### PR DESCRIPTION
**Pull request by AbdBarho**
clean up code.

The same should be done for evaluate tasks.

### CODE REVIEW

File: website/src/hooks/tasks/useGenericTaskAPI.tsx [hunk 1]
```diff
@@ -26,10 +26,10 @@ export const useGenericTaskAPI = <TaskType,>(taskApiEndpoint: string) => {
   );
 
   useEffect(() => {
-    if (tasks.length === 0 && !isLoading && !error) {
+    if (data === undefined && !isLoading && !error) {
```
Comment (Klotske) [inline website/src/hooks/tasks/useGenericTaskAPI.tsx:29]: Seems that change `undefined` broke fix #393. 
demo

Comment (AbdBarho) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx:29]: oh, how to reproduce?

Comment (AbdBarho) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: should be fixed now, could you please check again?

Comment (Klotske) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: Tested. Fix is working!

Comment (Klotske) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: Hmm. It's working 90% of the time. But when rapidly navigating it seems broken.
output3

Comment (Klotske) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: Maybe we should leave the check as it was? And wait for #551 to solve it?

Comment (AbdBarho) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: I reduced the dedupe time to half a second, which should be reasonable.

#551 might solve this entirely, I don't know if we should squeeze it before MVP.

Comment (fozziethebeat) [reply website/src/hooks/tasks/useGenericTaskAPI.tsx]: Let's get the random task change done then we can switch to `getServerSideProps`.  That can be rolled out anytime.

**Review by fozziethebeat — APPROVED** [review]

**Pull request merged**